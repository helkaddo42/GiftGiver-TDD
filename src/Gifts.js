import React, {Component} from 'react'
import { FormGroup,  FormControl, Form, Button} from 'react-bootstrap'

class Gifts extends Component {
    
    state={
        person: '',
        present: '',
    }
    render(){

        return (
            <div className='gifts'>
                <Form>
                    <FormGroup>
                        <label>Person</label>
                        <FormControl
                            className='input-person'
                            onChange={event=> this.setState({person: event.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <label>Present</label>
                        <FormControl
                            className='input-present'
                            onChange={event=> this.setState({present: event.target.value})}
                        />
                    </FormGroup>
                    <Button className='btn-remove' onClick={()=>this.props.removeGifts(this.props.gifts)} >Remove Gift</Button>

                </Form>
            </div>
        )

    }
    
}
export default Gifts