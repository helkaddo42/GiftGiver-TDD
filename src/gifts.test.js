import React from 'react';
import { shallow } from 'enzyme';
import Gifts from './Gifts';


describe('gifts',()=>{
    const wrapper =  shallow(<Gifts />);

    it(' gifts component exists',()=>{
        expect(wrapper).toMatchSnapshot();
    });

    it('state present && person exist',()=>{
        expect(wrapper.state()).toEqual({person: '', present: ''})
    })

});


describe('typing into person input',()=>{
    const wrapper =  shallow(<Gifts />);
   
    const person = 'mama'  
    
    beforeEach(()=>{
        wrapper.find('.input-person').simulate('change',{ target:{value: person}});
    })

    it('update the state person',()=>{
        expect(wrapper.state().person).toEqual(person);
    })
})

describe('typing into input present',()=>{
    const wrapper =  shallow(<Gifts />);
   
    const present = 'playstation'  
    
    beforeEach(()=>{
        wrapper.find('.input-present').simulate('change',{ target:{value: present}});
    })

    it('update the state present',()=>{
        expect(wrapper.state().present).toEqual(present);
    })
})

describe('remove gift',()=>{

    const removeFn = jest.fn();
    const id = 1;
    const wrapper =  shallow(<Gifts gifts={id} removeGifts={removeFn} />);

    beforeEach(()=>{
        wrapper.find('.btn-remove').simulate('click')
    })

    it('call removeGifts fn',()=>{
        expect(removeFn).toHaveBeenCalledWith(id)
    })
        
})