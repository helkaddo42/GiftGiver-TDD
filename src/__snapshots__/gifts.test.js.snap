// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gifts  gifts component exists 1`] = `ShallowWrapper {}`;
